File: 수학(구현)/백준_6588_골드바흐의 추측/골드바흐의 추측.kt
import java.util.*

fun is_prime(n : Int): Boolean{
    for(i in 2..n){
        if(i*i >n) break;
        if(n %i ==0) return false
    }
    return true
}

fun main(args: Array<String>){
    val s = Scanner(System.`in`)
    var a = 0
    while(true){
        var n = s.nextInt()
        if(n == 0) break

        for(i in 3..n-1 step 2){
            if(is_prime(i) && is_prime(n-i)){
                a = i; break
            }
        }
        if(a == 0) println("Goldbach's conjecture is wrong.");
        else println("${n} = ${a} + ${n-a}")
    }

}

File: 수학(구현)/백준_6588_골드바흐의 추측/골드바흐의 추측.cpp
#include <iostream>
using namespace std;

bool is_prime(int n)
{
    for (int i = 2; i * i <= n; i++)
    {
        if (n % i == 0)
            return false;
    }
    return true;
}

int main()
{
    int n;
    int a = 0;

    while (true)
    {
        scanf("%d", &n);
        if (n == 0)
            break;

        for (int i = 3; i < n; i += 2)
        { //홀수만
            if (is_prime(i) && is_prime(n - i))
            { //소수만
                a = i;
                break;
            }
        }

        if (a != 0)
            printf("%d = %d + %d\n", n, a, n - a);
        else
            printf("Goldbach's conjecture is wrong.");
    }
}
